alternator/executor.cc

```diff
@@ -1143,20 +1143,34 @@ future<executor::request_return_type> executor::update_table(client_state& clien
 // collection to be sorted by key order.
 class attribute_collector {
     std::map<bytes, atomic_cell, serialized_compare> collected;
+    gc_clock::duration _default_ttl;
+
     void add(bytes&& name, atomic_cell&& cell) {
         collected.emplace(std::move(name), std::move(cell));
     }
     void add(const bytes& name, atomic_cell&& cell) {
         collected.emplace(name, std::move(cell));
     }
 public:
-    attribute_collector() : collected(attrs_type()->get_keys_type()->as_less_comparator()) { }
+    explicit attribute_collector(gc_clock::duration default_ttl)
+        : collected(attrs_type()->get_keys_type()->as_less_comparator()),
+         _default_ttl{default_ttl}
+    {}
+
     void put(bytes&& name, const bytes& val, api::timestamp_type ts) {
-        add(std::move(name), atomic_cell::make_live(*bytes_type, ts, val, atomic_cell::collection_member::yes));
+        if (_default_ttl == gc_clock::duration::zero()) {
+            add(std::move(name), atomic_cell::make_live(*bytes_type, ts, val, atomic_cell::collection_member::yes));
+        } else {
+            add(std::move(name), atomic_cell::make_live(*bytes_type, ts, val, gc_clock::now() + _default_ttl, _default_ttl, atomic_cell::collection_member::yes));
+        }
 
     }
     void put(const bytes& name, const bytes& val, api::timestamp_type ts) {
-        add(name, atomic_cell::make_live(*bytes_type, ts, val, atomic_cell::collection_member::yes));
+        if (_default_ttl == gc_clock::duration::zero()) {
```

Comment:
yeah, it crossed my mind but I decided to wait for a review first (in order not to waste time on deduplication in case somebody totally opposes this series), I'll do that in v2